tests/legacy/unit/commands/test_upload_metadata.py

```diff
@@ -198,8 +198,10 @@ def json(self):
 
         raised = self.assertRaises(
             storeapi.errors.StoreUploadError,
-            self.run_command,
-            ["upload-metadata", self.snap_file],
+            lambda: self.run_command(
+                ["upload-metadata", self.snap_file],
+                input="n\n"
+            )
```

Comment:
There is a bunch of mocking and patching surrounding click in the legacy unit tests.

It seems something changed in click and the [confirm](https://github.com/canonical/snapcraft/blob/a3f33196ff0849107ff28b001c1caa2d7fc31175/snapcraft_legacy/cli/echo.py#L126) call was raising an [abort](https://github.com/pallets/click/blob/2d610e36a429bfebf0adb0ca90cdc0585f296369/src/click/termui.py#L239) instead defaulting to "no".